components/script/dom/htmlinputelement.rs

```diff
@@ -63,16 +63,118 @@ const PASSWORD_REPLACEMENT_CHAR: char = '●';
 #[derive(Clone, Copy, JSTraceable, PartialEq)]
 #[allow(dead_code)]
 #[derive(MallocSizeOf)]
-enum InputType {
-    InputSubmit,
-    InputReset,
-    InputButton,
-    InputText,
-    InputFile,
-    InputImage,
-    InputCheckbox,
-    InputRadio,
-    InputPassword
+pub enum InputType {
+    Button,
+    Checkbox,
+    Color,
+    Date,
+    Datetime,
```

Comment:
Do we really need this? `datetime` is [an obsolete value](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Input/datetime) and is not in the spec. It's also unsupported in Firefox. Doesn't seem like it needs to exist in Servo?

Reply:
Agreed. We can file a followup issue.